### PR TITLE
Add filter package

### PR DESCRIPTION
**Pull request by recrtl**
- New package to filter slices based on arbitrary rules
- JSON and query values support

### CODE REVIEW

File: pkg/filter/evaluators.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
+package filter
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+const (
```
Comment (mgiaccone) [inline pkg/filter/evaluators.go:9]: Why don't we use standard wording for this? Equal and NotEqual?

**Review by mgiaccone — APPROVED** [review]

File: pkg/filter/evaluators.go (hunk 1 above)
Comment (recrtl) [reply pkg/filter/evaluators.go:9]: Thanks for the suggestion :+1:

File: pkg/filter/evaluators.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+	return false
+}
+
+// nolint: gocyclo
+func convertValues(value interface{}) interface{} {
+	switch value := value.(type) {
+	case int:
+		return float64(value)
+	case int8:
```
Comment (BoRuDar) [inline pkg/filter/evaluators.go:72]: you may put it all in a one line like `case int, int8...:` etc

**Review by nicolaasuni — APPROVED** [review]

File: pkg/filter/evaluators.go (hunk 2 above)
Comment (nicolaasuni) [reply pkg/filter/evaluators.go:72]: Doing that will result in:
"cannot convert value (variable of type interface{}) to float64 (need type assertion) (typecheck)"

Comment (nicolaasuni) [reply pkg/filter/evaluators.go:72]: I have seen this before, let me check what the actual solution was

Comment (nicolaasuni) [reply pkg/filter/evaluators.go:72]: It looks like we have to keep these separate entries for the proper type assertion.

**Pull request merged**